src/tools/perf/lib/uct_tests.cc

```diff
@@ -398,16 +430,26 @@ class uct_perf_test_runner {
             } else{
                 send_sn     = 0; /* Remote buffer will remain 0 throughout the test */
             }
-            *(psn_t*)buffer = send_sn;
+
+            set_sn(&m_perf, (psn_t*)buffer, m_perf.uct.send_mem.mem_type,
+                   &send_sn, UCS_MEMORY_TYPE_HOST);
 
             UCX_PERF_TEST_FOREACH(&m_perf) {
                 if (flow_control) {
                     /* Wait until getting ACK from responder */
-                    ucs_assertv(UCS_CIRCULAR_COMPARE8(send_sn - 1, >=, *recv_sn),
-                                "recv_sn=%d iters=%ld", *recv_sn, m_perf.current.iters);
-                    while (UCS_CIRCULAR_COMPARE8(send_sn, >, (psn_t)(*recv_sn + fc_window))) {
+                    sn = get_recv_sn(&m_perf,
+                                     const_cast<const psn_t*>(recv_sn),
```

Comment:
const_cast should not be needed in this case. its use is to remove const qualifier. here we don't strip the const.

Reply:
`const_cast` is used here to remove `volatile` qualifier